frontend/src/api/geocaches.ts

```diff
@@ -7,3 +7,9 @@ export const getGeoCaches = async (filters?: Filters) => {
     return geocaches.data as Array<Geocache>;
 };
 
+export const getCache = async (id?: string | undefined) => {
+    if(!(id)) { id = "null"; }
+    const cache = (await getGeoCaches()).filter(item => item.referenceCode==id);
+    if(!cache || cache.length < 1) return {} as Geocache;
+    return cache[0] as Geocache; 
+};
```

Comment:
The function should only take string as a valid parameter. It makes no sense to allow undefined as a function parameter. Also we should call a different endpoint to only get a single geocache, not do it like this with filtering. It's highly inefficient. Here's a fixed version:


export const getCache = async (id: string) => {
    const response = await axios.get(`/api/geocaches/${id}`);
    return response.data as Geocache; 
};

Reply:
Ok